enhanced/scraper.go

```diff
@@ -87,20 +92,30 @@ func (s *scraper) scrape(ctx context.Context) map[string][]prometheus.Metric {
 			l = l.With("region", instance.Region).With("instance", instance.Instance)
 
 			// l.Debugf("Message:\n%s", *event.Message)
-			osMetrics, err := parseOSMetrics([]byte(*event.Message))
+			osMetrics, err := parseOSMetrics([]byte(*event.Message), s.testDisallowUnknownFields)
 			if err != nil {
+				// only for tests
+				if s.testDisallowUnknownFields {
```

Comment:
why do we need it? How can it help with tests?

Reply:
This field is set to `true` in tests. When there are new metrics in CloudWatch Logs, tests will fail.
See discussion at https://jira.percona.com/browse/PMM-3835 for some more background.